src/reader.rs

```diff
@@ -6511,10 +6511,22 @@ impl<'a> Reader<'a> {
             prefix.push_utfstr(&tok);
             prefix.push_utfstr(&common_prefix);
         } else {
-            // Append just the end of the string.
+            // Collapse parent directories and append end of string
             prefix.push(get_ellipsis_char());
+
             let full = tok + &common_prefix[..];
-            prefix.push_utfstr(&full[full.len() - PREFIX_MAX_LEN..]);
+            let truncated = &full[full.len() - PREFIX_MAX_LEN..];
+            let parts: Vec<&wstr> = truncated.split('/').collect();
+            if parts.len() < 2 {
+                // No path separators were found in the common prefix, so we can't collapse
+                // any further
+                prefix.push_utfstr(&truncated);
+            } else {
+                // Discard any parent directories and include whats left
```

Comment:
this works for absolute paths but it doesn't seem to work for `~/some/path`, can we fix that?

Reply:
Unless I'm misunderstanding you, it should already work for relative directories, but it depends on the length of the prefix. For example, here's my fish config dir:
![image](https://github.com/user-attachments/assets/0652db59-1ba9-45c3-bd9e-09d2cc1e9f96)

But even with an absolute path, a short prefix won't collapse:
![image](https://github.com/user-attachments/assets/f0c94eec-1aba-472b-94f2-38ff92a29c74)

Reply:
yeah my bad